paymentsheet/src/main/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethod.kt

```diff
@@ -0,0 +1,50 @@
+package com.stripe.android.paymentsheet.model
+
+import android.os.Parcelable
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.paymentsheet.specifications.bancontact
+import com.stripe.android.paymentsheet.specifications.sofort
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Enum defining all payment methods supported in Payment Sheet.
+ */
+@Parcelize
+enum class SupportedPaymentMethod(
```

Comment:
sweet, and then one more to move these into the spec?

Reply:
Into `FormSpec`? Not sure what you have in mind, but we need an enum or a map to figure out the intersection between the payment methods supported on the `StripeIntent` and the ones Payment Sheet supports.

Reply:
What if FormSpec becomes SupportedPaymentMethodSpec, and this contains the definition of a single payment method type.   Then you can have SupportedPaymentMethodRepository which has the map of String which is in the paymentIntent and the SupportedPaymentMethodSpec.

Reply:
That works, as long as we have a Parcelable that's passed on to the Fragment and is used to identify the payment method.